### PR TITLE
Add golangci-lint to CI

### PR DESCRIPTION
**Pull request by sonatard**
I disable any rules in https://github.com/lestrrat-go/jwx/pull/166/commits/b75330f749e8dca19f5cd6b926181a50a62e2d32. if you think need it, I'll fix it.

And `enable-all: true` is so strong rule. If you don't like it please tell me what disable rules.

### CODE REVIEW

File: jwe/internal/cmd/genheader/main.go [hunk 1]
```diff
@@ -3,14 +3,15 @@ package main
...
 	"bytes"
 	"fmt"
-	"go/format"
 	"log"
 	"os"
 	"sort"
 	"strconv"
 	"strings"
 
+	"github.com/lestrrat-go/jwx/jwe"
```
Comment (lestrrat) [inline jwe/internal/cmd/genheader/main.go:12]: no, don't import from the package that we're generating into. I know this causes problems

File: jws/internal/cmd/genheader/main.go [hunk 2]
```diff
@@ -3,14 +3,15 @@ package main
...
 	"bytes"
 	"fmt"
-	"go/format"
 	"log"
 	"os"
 	"sort"
 	"strconv"
 	"strings"
 
+	"github.com/lestrrat-go/jwx/jws"
```
Comment (lestrrat) [inline jws/internal/cmd/genheader/main.go:12]: same here. no importing from the package that we're generating

**Comment by sonatard** [comment]
Thanks for review! I fixed it.

**Comment by lestrrat** [comment]
Thank you very much!

**Pull request merged**